stacks-signer/src/runloop.rs

```diff
@@ -300,10 +345,177 @@ impl<C: Coordinator> RunLoop<C> {
         }
     }
 
+    /// Helper function to validate a signature share request, updating its message where appropriate.
+    /// If the request is for a block it has already agreed to sign, it will overwrite the message with the agreed upon value
+    /// Returns whether the request is valid or not.
+    fn validate_signature_share_request(&self, request: &mut SignatureShareRequest) -> bool {
+        // A coordinator could have sent a signature share request with a different message than we agreed to sign
+        match self
+            .blocks
+            .get(&request.message)
+            .map(|block_info| &block_info.vote)
+        {
+            Some(Some(vote)) => {
+                // Overwrite with our agreed upon value in case another message won majority or the coordinator is trying to cheat...
+                request.message = vote.clone();
+                true
+            }
+            Some(None) => {
+                // We have seen this block before, but we have not agreed to sign it. TODO: ignore it
+                debug!("Received a signature share request for a block we have not validated yet.");
+                false
+            }
+            None => {
+                // We have not seen this block before.
+                // TODO: should probably ignore any messages that are not either sBTC transactons or Nakamoto blocks. Leave now for abitrary message signing
```

Comment:
Isn't this a problem though?  We don't want the system to offer a signing oracle

Reply:
I might be able to update these tests now...will see...